client/main.lua

```diff
@@ -226,14 +226,24 @@ end
 ---@param args table<string, any>
 local function openVehicleCategoryMenu(args)
     local categoryMenu = {}
+    local sortedCategories = {}
     for k, v in pairs(config.shops[insideShop].categories) do
-        categoryMenu[#categoryMenu + 1] = {
-            title = v,
-            arrow = true,
-            onSelect = function()
-                openVehCatsMenu(k, args.targetVehicle)
+        table.insert(sortedCategories, v)
+    end
+    table.sort(sortedCategories)
+    for _, v in ipairs(sortedCategories) do
+        for k, originalValue in pairs(config.shops[insideShop].categories) do
+            if v == originalValue then
+                categoryMenu[#categoryMenu + 1] = {
+                    title = v,
+                    arrow = true,
+                    onSelect = function()
+                        openVehCatsMenu(k, args.targetVehicle)
+                    end
+                }
+                break
             end
-        }
+        end
```

Comment:
I'm confused as to the purpose of the inner for loop. Can't the category menu be created by sortedCategories only?

Reply:
Honestly. I was kinda just winging it. But I'm more experianced in Glua not regular lua so some of my code might be off. 
Ive recoded it somewhat since the original ive posted here and it seems to work for me no issues

Reply:
Yeah this would work, but it's more complex than it needs to be. I think you can remove the inner for loop and create categoryMenus inside the outer for loop.